pages/articles/[slug].tsx

```diff
@@ -7,27 +7,50 @@ import { GetStaticPaths, GetStaticProps } from "next";
 import { getArticleBySlug, getArticlesForListing, getSiteMenu } from "../../lib/kontentClient";
 import { RichTextElement } from "../../components/shared/RichTextContent";
 import { mainColorBgClass } from "../../lib/constants/colors";
-import { siteCodename } from "../../lib/utils/env";
+import { siteCodename } from '../../lib/utils/env';
+import { formatDate } from "../../lib/utils/dateTime";
+import { AuthorHorizontal } from "../../components/shared/AuthorHorizontal";
+import { AuthorVertical } from "../../components/shared/AuthorVertical";
+import { ProductCategory } from '../../models/taxonomies/productCategory';
+import { useSiteCodename } from "../../components/shared/siteCodenameContext";
 
 type Props = Readonly<{
   article: Article;
   siteCodename: ValidCollectionCodename;
   siteMenu?: Block_Navigation;
 }>;
 
-const ArticlePage: FC<Props> = props => (
-  <AppPage siteCodename={props.siteCodename} siteMenu={props.siteMenu}>
-    <HeroImage url={props.article.elements.heroImage.value[0]?.url} itemId={props.article.system.id}>
-      <div className={`py-1 px-3 w-full md:w-fit ${mainColorBgClass[props.siteCodename]}  opacity-[85%]`}>
-        <h1 className="m-0 text-3xl tracking-wide font-semibold">{props.article.elements.title.value}</h1>
+const ArticlePage: FC<Props> = props => {
+  const siteCodename = useSiteCodename();
+  return (
+    <AppPage siteCodename={props.siteCodename} siteMenu={props.siteMenu}>
+      <HeroImage url={props.article.elements.heroImage.value[0]?.url} itemId={props.article.system.id}>
+        <div className={`py-1 px-3 w-full md:w-fit ${mainColorBgClass[props.siteCodename]}  opacity-90`}>
+          <h1 className="m-0 text-3xl tracking-wide font-semibold">{props.article.elements.title.value}</h1>
+        </div>
+        <div className="bg-white opacity-90 p-4">
+          <p className="font-semibold">
+            {props.article.elements.abstract.value}
+          </p>
+        </div>
+      </HeroImage>
+      <div className="max-w-screen-md m-auto">
+        <AuthorHorizontal item={props.article.elements.author.linkedItems[0]} />
+        <div className="flex flex-col gap-2">
+          <div className="w-fit p-2 bg-gray-800 text-white opacity-90 font-semibold">{props.article.elements.publishingDate.value && formatDate(props.article.elements.publishingDate.value)}</div>
+          <div className="flex gap-2" >
+            {
+              props.article.elements.articleType.value.length > 0 && props.article.elements.articleType.value.map(type => (
+                <div key={type.codename} className={`w-fit p-2 ${mainColorBgClass[siteCodename]} font-semibold`}>{type.name}</div>
```

Comment:
This is kind of weird
`mainColorBgClass[siteCodename]` does not really work.

![image](https://github.com/kontent-ai-bot/kontent-ai-new-sample-app/assets/54802833/ac0e495d-82bb-4257-9344-2b94c7b141cb)

However, ArticlePage has as a props siteCodename which is also used in line 28. With that one it works properly.

A quick fix is to change `siteCodename` to `props.siteCodename.`

However, it is weird that useSitecodename return wrong codename

Reply:
Also there is no space between `Name` and `Surname` in `authorHorizontal`

Reply:
Fixed both - it has something to do what is rendered on server and what on client and when

Reply:
There is default value being set: `const SiteContext = createContext<CodenameContext>({ siteCodename: "ficto_healthtech" });
`